### PR TITLE
chore(taiko-client): add hive tests to workflow

### PR DESCRIPTION
**Pull request by mask-pp**
run `make hive_test`
```
sh integration_test/hive_test.sh
```

equal to

```
git clone git@github.com:taikoxyz/hive.git
cd hive && make hive
./build/bin/hive --client taiko2/geth,taiko2/prysm-bn,taiko2/prysm-vc --sim taiko2 --sim.limit "eth2-deneb-testnet/test-deneb-genesis"
```

### CODE REVIEW

File: .github/workflows/taiko-client--docker.yml [hunk 1]
```diff
@@ -51,3 +51,7 @@ jobs:
           push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+
+      - name: hive test
```
Comment (RogerLamTd) [inline .github/workflows/taiko-client--docker.yml:55]: could we make this change in a separate PR and mark it with feat(repo): ?

Comment (mask-pp) [reply .github/workflows/taiko-client--docker.yml:55]: When a new Docker image is generated, the Hive test needs to be triggered. I haven't found an effective way to achieve this yet. Do you have any better suggestions?

File: .github/workflows/taiko-client--docker.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: "Push docker image to GAR"
 
 on:
   push:
-    branches: [main]
+    branches: [hive_tests_]
```
Comment (mask-pp) [inline .github/workflows/taiko-client--docker.yml:5]: Temporarily change it in order to trigger the hive test.

File: .github/workflows/taiko-client--docker.yml (hunk 1 above)
Comment (RogerLamTd) [reply .github/workflows/taiko-client--docker.yml:55]: We can keep this solution, I just want to migrate the change to a separate PR for documentation clarity!

Comment (mask-pp) [reply .github/workflows/taiko-client--docker.yml]: `taiko-client--hive_test.yml` is added.

File: packages/taiko-client/integration_test/hive_tests.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+)
+
+func TestHiveHandler(t *testing.T) {
+	baseDir := os.Getenv("HIVE_BASE_DIR")
+	if baseDir == "" {
+		t.SkipNow()
+	}
+	handler, err := hivesim.NewHiveFramework(&hivesim.HiveConfig{
+		BaseDir:        baseDir,
+		SimPattern:     "taiko2",
```
Comment (davidtaikocha) [inline packages/taiko-client/integration_test/hive_tests.go:19]: Let's use `taiko` instead

Comment (mask-pp) [reply packages/taiko-client/integration_test/hive_tests.go:19]: OK, will be fixed in the hive repo and then change here.

**Review by davidtaikocha — APPROVED** [review]

**Review by RogerLamTd — APPROVED** [review]
